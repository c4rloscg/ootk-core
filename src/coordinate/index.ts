export * from './ClassicalElements';
export * from './EquinoctialElements';
export * from './FormatTle';
export * from './Geodetic';
export * from './ITRF';
export * from './J2000';
export * from '../enums/OrbitRegime';
export * from './RelativeState';
export * from './RIC';
export * from './StateVector';
export * from './TEME';
export * from './Tle';
