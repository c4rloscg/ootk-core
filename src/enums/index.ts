export { AngularDiameterMethod } from './AngularDiameterMethod';
export { AngularDistanceMethod } from './AngularDistanceMethod';
export { PassType } from './PassType';
