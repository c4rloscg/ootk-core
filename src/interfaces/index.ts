export type { BaseObjectParams } from './BaseObjectParams';
export type { GroundPositionParams } from './GroundPositionParams';
export type { OptionsParams } from './OptionsParams';
export type { SatelliteParams } from './SatelliteParams';
export type { SensorParams } from './SensorParams';
export type { StarObjectParams } from './StarObjectParams';
export type { ClassicalElementsParams } from './ClassicalElementsParams';
export type { EquinoctialElementsParams } from './EquinoctialElementsParams';
