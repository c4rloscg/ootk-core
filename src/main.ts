/**
 * @author Theodore Kruczek.
 * @license MIT
 * @copyright (c) 2022-2024 Theodore Kruczek Permission is
 * hereby granted, free of charge, to any person obtaining a copy of this
 * software and associated documentation files (the "Software"), to deal in the
 * Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
 * the Software, and to permit persons to whom the Software is furnished to do
 * so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 * @copyright Copyright (c) 2023 David RC Dayton
 *
 * Many of the classes are based off of the work of @david-rc-dayton and his
 * Pious Squid library
 * @see https://github.com/david-rc-dayton/pious_squid
 */

export * from './enums';
export * from './types/types';
export * from './interfaces';
export * from './time/time';
export * from './transforms';
export * from './utils';
export * from './operations/operations';
export { BaseObject } from './objects';
export { Earth } from './body';
export * from './coordinate';
export * from './observation';
export * from './data/DataHandler';
export * from './sgp4';
export * from './objects';
export * from './body';
