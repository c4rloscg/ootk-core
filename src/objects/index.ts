export { BaseObject } from './BaseObject';
export { Satellite } from './Satellite';
export { Sensor } from './Sensor';
export { Star } from './Star';
