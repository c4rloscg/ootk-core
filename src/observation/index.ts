export { RadecGeocentric } from './RadecGeocentric';
export { RadecTopocentric } from './RadecTopocentric';
export { RAE } from './RAE';
