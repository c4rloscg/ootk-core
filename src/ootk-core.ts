/**
 * @author @thkruz Theodore Kruczek
 * @description Orbital Object ToolKit Core (OOTK-Core) is a base collection
 * of tools for working with satellites and other orbital objects. This core
 * library is provided under the MIT license and is free to use in any project.
 * For additional features, see the full library at:
 * https://github.com/thkruz/ootk.
 *
 * @license MIT License
 * @Copyright (c) 2020-2024 Theodore Kruczek
 *
 * Many of the classes are based off of the work of @david-rc-dayton and his
 * Pious Squid library (https://github.com/david-rc-dayton/pious_squid).
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

export * from './body';
export * from './coordinate';
export * from './data/DataHandler';
export * from './enums';
export * from './interfaces';
export * from './objects';
export * from './observation';
export * from './operations/operations';
export { Sgp4 } from './sgp4/sgp4';
export * from './time/time';
export { Tle } from './tle/tle';
export * from './transforms';
export * from './types/types';
export * from './utils/constants';
export * from './utils/functions';
export { Utils } from './utils/utils';
