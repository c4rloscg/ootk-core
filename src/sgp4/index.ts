export { Sgp4 } from './sgp4';
