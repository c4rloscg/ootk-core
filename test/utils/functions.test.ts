import { Radians, acoth, acsch, asech, concat, copySign, csch, log10, sech, wrapAngle } from '../../src/main';

describe('functions', () => {
  it('should be calculate log10', () => {
    expect(log10(100)).toMatchSnapshot();
  });

  // sech
  it('should be calculate sech', () => {
    expect(sech(1)).toMatchSnapshot();
  });

  // csch
  it('should be calculate csch', () => {
    expect(csch(1)).toMatchSnapshot();
  });

  // acsch
  it('should be calculate acsch', () => {
    expect(acsch(1)).toMatchSnapshot();
  });

  // asech
  it('should be calculate asech', () => {
    expect(asech(1)).toMatchSnapshot();
  });

  // acotch
  it('should be calculate acotch', () => {
    expect(acoth(1)).toMatchSnapshot();
  });

  // copySign
  it('should be calculate copySign', () => {
    expect(copySign(1, 1)).toMatchSnapshot();
    expect(copySign(-1, 1)).toMatchSnapshot();
    expect(copySign(1, -1)).toMatchSnapshot();
    expect(copySign(-1, -1)).toMatchSnapshot();
  });

  // concat
  it('should be calculate concat', () => {
    expect(concat(new Float64Array([1, 2, 3]), new Float64Array([4, 5, 6]))).toMatchSnapshot();
  });

  // wrapAngle
  it('should be calculate wrapAngle', () => {
    expect(wrapAngle(1 as Radians)).toMatchSnapshot();
    expect(wrapAngle(5 as Radians)).toMatchSnapshot();
    expect(wrapAngle(-5 as Radians)).toMatchSnapshot();
  });

});
