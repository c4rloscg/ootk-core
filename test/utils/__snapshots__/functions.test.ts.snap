// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions should be calculate acotch 1`] = `Infinity`;

exports[`functions should be calculate acsch 1`] = `0.8813735870195429`;

exports[`functions should be calculate asech 1`] = `0`;

exports[`functions should be calculate concat 1`] = `
Float64Array [
  1,
  2,
  3,
  4,
  5,
  6,
]
`;

exports[`functions should be calculate copySign 1`] = `1`;

exports[`functions should be calculate copySign 2`] = `1`;

exports[`functions should be calculate copySign 3`] = `-1`;

exports[`functions should be calculate copySign 4`] = `-1`;

exports[`functions should be calculate csch 1`] = `0.8509181282393216`;

exports[`functions should be calculate log10 1`] = `2`;

exports[`functions should be calculate sech 1`] = `0.6480542736638855`;

exports[`functions should be calculate wrapAngle 1`] = `1`;

exports[`functions should be calculate wrapAngle 2`] = `-1.2831853071795862`;

exports[`functions should be calculate wrapAngle 3`] = `-5`;
